sdk/core/core-lro/src/lroEngine/operation.ts

```diff
@@ -0,0 +1,130 @@
+   *      send final GET request and return result from operation. If no final GET
+   *      is required, just return the result from operation.
+   *      - Determining what to call for final request is responsibility of each strategy
+   *  2.2 If it is not terminal state, call the polling operation and go to step 1
+   *      - Determining what to call for polling is responsibility of each strategy
+   *      - Strategies will always use the latest URI for polling if provided otherwise
+   *        the last known one
+   */
+  async update(options?: {
+    abortSignal?: AbortSignalLike | undefined;
+    fireProgress?: ((state: TState) => void) | undefined;
+  }): Promise<PollOperation<TState, TResult>> {
+    const state = this.state;
+    if (!state.isStarted) {
+      const initializeState = createInitializeState(
+        state,
+        this.lro.requestPath,
+        this.lro.requestMethod
+      );
+      await this.lro.sendInitialRequest(initializeState);
+    }
+
+    if (!state.isCompleted) {
+      if (this.poll === undefined || this.getLroStatusFromResponse === undefined) {
+        if (state.config === undefined) {
+          throw new Error(
+            "Bad state: LRO mode is undefined. Please check if the serialized state is well-formed."
+          );
+        }
+        this.getLroStatusFromResponse = createGetLroStatusFromResponse(
+          this.lro,
+          state.config,
+          this.lroResourceLocationConfig
+        );
+        this.poll = createPoll(this.lro);
+      }
+      if (state.pollingURL === undefined) {
+        throw new Error(
+          "Bad state: polling URL is undefined. Please check if the serialized state is well-formed."
+        );
+      }
+      const memoizedGetLroStatusFromResponse = memoize(this.getLroStatusFromResponse)();
+      const currentState = await this.poll(
+        state.pollingURL,
+        this.pollerConfig!,
+        memoizedGetLroStatusFromResponse
+      );
+      if (currentState.done) {
+        state.result = currentState.flatResponse;
+        state.isCompleted = true;
+      } else {
+        this.poll = currentState.next ?? this.poll;
+        state.pollingURL = getPollingUrl(currentState.rawResponse, state.pollingURL);
+      }
+    }
+    options?.fireProgress?.(state);
+    return this;
+  }
+
+  async cancel(): Promise<PollOperation<TState, TResult>> {
+    this.state.isCancelled = true;
+    return this;
+  }
+
+  /**
+   * Serializes the Poller operation.
+   */
+  public toString(): string {
+    return JSON.stringify({
+      state: this.state
+    });
+  }
+}
+
+function memoize<TInput extends unknown[], TResult>(
+  f: (...args: TInput) => TResult
+): () => (...args: TInput) => TResult {
+  return (): ((...args: TInput) => TResult) => {
+    let result: TResult | undefined = undefined;
+    return (...args: TInput): TResult => {
+      if (result === undefined) {
+        result = f(...args);
+      }
+      return result;
+    };
+  };
```

Comment:
I feel like there is an extra level in here that I don't understand. 

First you pass it a function to memoize, then you call it again to get the actual memoized function? It also doesn't seem to be checking the args at all, it just caches the last result regardless of if the args change?

I would expect it to work like this:

```ts
function add(a: number, b:number): number {
   return a + b;
}
const memoizedAdd = memoize(add);
memoizedAdd(1, 2); // returns 3 after calling add
memoizedAdd(1, 2); // returns 3 without calling add
memoizedAdd(3, 4); // returns 7 after calling add
```

Reply:
I just did not go all the way on memoization for each input because I did not need it but it is simple enough I could as well just add it. Also, you're right, I accidentally introduced an extra outer function.

Reply:
Done.